client/jetpack-cloud/sections/partner-portal/issue-license-form/index.tsx

```diff
@@ -39,11 +42,40 @@ export default function IssueLicenseForm( {
 	const products = useProductsQuery( {
 		select: alphabeticallySortedProductOptions,
 	} );
+
+	const fromDashboard = getQueryArg( window.location.href, 'source' ) === 'dashboard';
+
 	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const [ product, setProduct ] = useState( '' );
+
+	const handleRedirectToDashboard = useCallback(
+		( licenseKey ) => {
+			if ( selectedSite ) {
+				const selectedProduct = products?.data?.find( ( p ) => p.slug === product );
+				if ( selectedProduct ) {
+					dispatch(
+						setPurchasedProduct( {
+							selectedSite: selectedSite.domain,
+							selectedProduct: {
+								name: getProductTitle( selectedProduct.name ),
+								key: licenseKey,
+							},
+						} )
+					);
+					return page.redirect( '/dashboard' );
+				}
+			}
```

Comment:
IMO flipping the conditions and using early returns to reduce the indentation would make this easier to read.